src/components/TopicArticles.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from "react";
+import { fetchArticles } from "../Utils/api";
+import { Link } from "react-router-dom";
+
+function TopicArticles({ selectedTopic }) {
+  console.log("selectedTopic", selectedTopic);
+  const [articles, setArticles] = useState([]);
+
+  useEffect(() => {
+    fetchArticles().then((data) => {
+      console.log(data);
+      const filteredArticles = data.filter(
```

Comment:
We should ideally be achieving the filtering of our articles by topic using our backend, we can pass optional queries through to axios whether topic is defined in the url or not.
This can all be done using the same articles component and getArticles api request, take a look at the docs on axios for help on this :)